src/core/namespace.c

```diff
@@ -385,6 +387,8 @@ int setup_namespace(
                 strv_length(read_only_paths) +
                 strv_length(inaccessible_paths) +
                 private_dev +
+                (protect_sysctl ? 3 : 0) +
+                (protect_cgroups != protect_sysctl) +
```

Comment:
I am not sure if this one is needed since mount paths are reordered later in `mount_path_compare()` by the most restrictive mode and we can always assert that the size of the mount array didn't overflow. So why not just test `protect_cgroups`  ? I'm missing something here...